builder/exoscale/step_download_snapshot.go

```diff
@@ -0,0 +1,88 @@
+package exoscale
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	"github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+type stepDownloadSnapshot struct {
+	builder *Builder
+}
+
+func (s *stepDownloadSnapshot) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	var (
+		snapshotURL      = state.Get("snapshot_url").(string)
+		snapshotChecksum = state.Get("snapshot_checksum").(string)
+		ui               = state.Get("ui").(packer.Ui)
+	)
+
+	if !s.builder.config.SnapshotDownload {
+		return multistep.ActionContinue
+	}
+
+	ui.Say("Downloading compute instance snapshot")
+
+	if err := s.downloadSnapshot(ui, snapshotURL); err != nil {
+		ui.Error(fmt.Sprintf("Unable to download compute instance snapshot: %v", err))
+		return multistep.ActionHalt
+
+	}
+	if err := s.createChecksumFile(snapshotChecksum); err != nil {
+		ui.Error(fmt.Sprintf("Unable to create checksum file of the snapshot: %v", err))
+		return multistep.ActionHalt
+	}
+
+	return multistep.ActionContinue
+}
+
+func (s *stepDownloadSnapshot) downloadSnapshot(ui packer.Ui, snapshotURL string) error {
+	templateFile := s.builder.config.TemplateName + ".template"
```

Comment:
```suggestion
	templateFile := s.builder.config.TemplateName + ".qcow2"
```
(hint at the actual type of the downloaded file)

Reply:
that's true and that's what I had in the beginning, however I was basing on your documentation :-) https://community.exoscale.com/documentation/compute/snapshots/#:~:text=wget%20%2DO%20ce295d39%2D421e%2D49c4%2D81f7%2D9890c3db3600.template